### PR TITLE
Added defensive checks for reseting iTime

### PR DESCRIPTION
**Pull request by bmabsout**
Made timing variables doubles and checked for nans

### CODE REVIEW

File: app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public class Shader {
 
     private int targetFPS = 30;
     private long startTime;
-    private int frame;
+    private unsigned int frame;
```
Comment (oxters168) [inline app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java:59]: Java does not have support for unsigned ints at the moment, please remove

File: app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java [hunk 2]
```diff
@@ -325,15 +325,16 @@ protected void draw() {
 
         frame++;
         long currentTime = System.currentTimeMillis() - startTime;
-        float secondsElapsed = (currentTime) / 1000f; // will be input into shader as iTime
-        float deltaTime = (currentTime - prevTime) / 1000f;
-        float fps = 1 / deltaTime;
+        double secondsElapsed = (currentTime) / 1000.0; // will be input into shader as iTime
+        double deltaTime = (currentTime - prevTime) / 1000.0;
+        double fps = 1 / deltaTime;
```
Comment (oxters168) [inline app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java:330]: Changing to double causes issue when passing the parameter to the shader, please switch back to float

File: app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java [hunk 3]
```diff
@@ -325,15 +325,16 @@ protected void draw() {
...
         prevTime = currentTime;
-        if (secondsElapsed > 60 * 60 * 24) {
+        if (secondsElapsed > 60 * 60 * 24 || Double.isNaN(fps) || Double.isNan(deltaTime)) {
```
Comment (oxters168) [inline app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java:332]: As per the previous comment, this should now be Float.isNaN. Also the second call needs the last n to be capitalized.

File: app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java [hunk 4]
```diff
@@ -325,15 +325,16 @@ protected void draw() {
...
             frame = 0;
             startTime = System.currentTimeMillis();
             prevTime = 0;
             secondsElapsed = 0;
+            fps = 60.0;
```
Comment (oxters168) [inline app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java:337]: This should now be 60f rather than 60.0

**Review by oxters168 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java (hunk 4 above)
Comment (bmabsout) [reply app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java:337]: Doesn't matter but I'll change it

File: app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java (hunk 3 above)
Comment (bmabsout) [reply app/src/main/java/com/OxGames/OxShell/Wallpaper/Shader.java:332]: KKKKKKKKK

**Comment by oxters168** [comment]
Looks good, thank you for contributing!

**Pull request merged**